src/NuGet.Core/NuGet.Packaging/PackageReaderBase.cs

```diff
@@ -500,7 +501,9 @@ protected static bool IsReferenceAssembly(string path)
 
             if (StringComparer.OrdinalIgnoreCase.Equals(extension, ".dll"))
             {
-                if (!path.EndsWith(".resources.dll", StringComparison.OrdinalIgnoreCase))
+                // Resource assembly suppose to be located in the culture folder.
+                // In other case proper assemblies that are ended with "resources" can be filtered.
+                if (!Regex.IsMatch(path, @"[\\\/](\w{2}|\w{2}[-]\w{2})[\\\/].+[.]resources[.]dll$", RegexOptions.IgnoreCase))
```

Comment:
Create a static `Regex`, so it doesn't create a new regex object and parse the expression for every dll in the package.

Reply:
Can I place the static regex in the  PackageReaderBase?

Reply:
`PackageReaderBase` is good.
`private static readonly Regex _resourceDllRegex = new Regex(...)`